fs/ensure_syslink_test.ts

```diff
@@ -0,0 +1,123 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+import { test } from "../testing/mod.ts";
+import {
+  assertEquals,
+  assertThrows,
+  assertThrowsAsync
+} from "../testing/asserts.ts";
+import { ensureSyslink, ensureSyslinkSync } from "./ensure_syslink.ts";
+import * as path from "./path/mod.ts";
+
+const testdataDir = path.resolve("fs", "testdata");
+
+test(async function ensureSyslinkIfItNotExist() {
+  const testDir = path.join(testdataDir, "link_file_1");
+  const testFile = path.join(testDir, "test.txt");
+
+  assertThrowsAsync(async () => {
+    await ensureSyslink(testFile, path.join(testDir, "test1.txt"));
+  });
+
+  assertThrowsAsync(async () => {
+    await Deno.stat(testFile).then(() => {
+      throw new Error("test file should exists.");
+    });
+  });
+});
+
+test(function ensureSyslinkSyncIfItNotExist() {
+  const testDir = path.join(testdataDir, "link_file_2");
+  const testFile = path.join(testDir, "test.txt");
+
+  assertThrows(() => {
+    ensureSyslinkSync(testFile, path.join(testDir, "test1.txt"));
+  });
+
+  assertThrows(() => {
+    Deno.statSync(testFile);
+    throw new Error("test file should exists.");
+  });
+});
+
+test(async function ensureSyslinkIfItExist() {
```

Comment:
add condition for Deno.platform == 'win' and add a assertThrow() to handle the no implemented.

Reply:
make sense.